### PR TITLE
Create Url Repository File Impl in storage package

### PR DESCRIPTION
**Pull request by Pinkavsky**
Create the methods in Url Repository File: 
- createUrlAlias
- deleteUrlAlias
- findUrlAlias
- getAllAliasesForUser
- putInMapByEmail
and other.

### CODE REVIEW

**Review by IDavydenko21 — APPROVED** [review]

File: src/main/java/edu/kpi/testcourse/storage/UrlRepositoryFileImpl.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+  /**
+   * Создает экземпляр.
+   */
+  @Inject
+  public UrlRepositoryFileImpl(JsonTool jsonTool, UrlShortenerConfig appConfig) {
+    this.jsonTool = jsonTool;
+    this.jsonFilePath = makeJsonFilePath(appConfig.storageRoot());
+    this.urlMapByAlias = readUrlsFromJsonDatabaseFile(jsonTool, this.jsonFilePath);
+    this.urlsMapByEmail = makeUrlsMapByEmail(urlMapByAlias);
+  }
```
Comment (Pinkavsky) [inline src/main/java/edu/kpi/testcourse/storage/UrlRepositoryFileImpl.java:47]: Creates an instance.

Comment (Pinkavsky) [inline src/main/java/edu/kpi/testcourse/storage/UrlRepositoryFileImpl.java:47]: Creates an instance.

Comment (Pinkavsky) [inline src/main/java/edu/kpi/testcourse/storage/UrlRepositoryFileImpl.java:47]: Creates an instance.

File: src/main/java/edu/kpi/testcourse/storage/UrlRepositoryFileImpl.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+  public synchronized void createUrlAlias(UrlAlias urlAlias) throws AliasAlreadyExist {
+    if (urlMapByAlias.containsKey(urlAlias.alias())) {
+      throw new AliasAlreadyExist();
+    }
+
+    urlMapByAlias.put(urlAlias.alias(), urlAlias);
+    putInMapByEmail(urlsMapByEmail, urlAlias);
+
+    syncUrlsWithJsonDatabaseFile();
+  }
```
Comment (Pinkavsky) [inline src/main/java/edu/kpi/testcourse/storage/UrlRepositoryFileImpl.java:71]: Creates an UrlAlias.

File: src/main/java/edu/kpi/testcourse/storage/UrlRepositoryFileImpl.java [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
+  private void syncUrlsWithJsonDatabaseFile() {
+    String json = jsonTool.toJson(urlMapByAlias);
+    try {
+      Files.writeString(jsonFilePath, json);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
```
Comment (IDavydenko21) [inline src/main/java/edu/kpi/testcourse/storage/UrlRepositoryFileImpl.java:142]: Synchronize urls with Json Database.

File: src/main/java/edu/kpi/testcourse/storage/UrlRepositoryFileImpl.java [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+    if (!urlAlias.email().equals(email)) {
+      throw new PermissionDenied();
+    }
+
+    urlMapByAlias.remove(alias);
+    List<UrlAlias> userUrls = urlsMapByEmail.get(email);
+    userUrls.remove(urlAlias);
+
+    syncUrlsWithJsonDatabaseFile();
+  }
```
Comment (Narberial) [inline src/main/java/edu/kpi/testcourse/storage/UrlRepositoryFileImpl.java:93]: Delete Url Alias

**Pull request merged**